datafusion/src/datasource/listing/table.rs

```diff
@@ -40,7 +44,70 @@ use crate::datasource::{
 
 use super::helpers::{expr_applicable_for_cols, pruned_partition_list, split_files};
 
+/// Configuration for creating a 'ListingTable'  
+pub struct ListingTableConfig {
+    pub object_store: Arc<dyn ObjectStore>,
+    pub table_path: String,
+    pub file_schema: Option<SchemaRef>,
+    pub options: Option<ListingOptions>,
+}
+
+impl ListingTableConfig {
+    /// Creates new `ListingTableConfig`.  The `SchemaRef` and `ListingOptions` are inferred based on the suffix of the provided `table_path`.
+    pub async fn new(
+        object_store: Arc<dyn ObjectStore>,
+        table_path: impl Into<String>,
+    ) -> Self {
+        Self {
+            object_store,
+            table_path: table_path.into(),
+            file_schema: None,
+            options: None,
+        }
+    }
+    pub fn with_schema(mut self, schema: SchemaRef) {
```

Comment:
If you consume self and return Self, in these APIs, then you will be able to chain them together for a better UX:

```suggestion
    pub fn with_schema(self, schema: SchemaRef) -> Self {
```

Something like this:

```rust
let cfg = ListingTableConfig::new().with_listing_options().with_scheame();
```

Reply:
Thx @houqp.  FYI, i made updates and im planning on updating the remaining errors next.  Lmk if you see anything else.

Reply:
Looks good to me :+1: